tsdoc/src/nodes/DocComment.ts

```diff
@@ -114,13 +116,27 @@ export class DocComment extends DocNode {
     return DocNodeKind.Comment;
   }
 
+  /**
+   * The collection of all `@see` DockBlockTag nodes belonging to this doc comment.
+   */
+  public get seeBlocks(): ReadonlyArray<DocBlock> {
+    return this._seeBlocks;
+  }
+
   /**
    * The collection of all DocBlock nodes belonging to this doc comment.
    */
   public get customBlocks(): ReadonlyArray<DocBlock> {
     return this._customBlocks;
   }
 
+  /**
+   * Append an item to the seeBlocks collection.
+   */
+  public appendSeeBlock(block: DocBlock): void {
```

Comment:
Hmm... this API feels weird to me. The original design is that the "core" tags (`Standardization.Core`) would get dedicated properties in the `DocComment` API, and then all other tags would go into the `customBlocks` array.

But this has a downside that any code relying on `customBlocks` will get broken by an update that promotes a block to being "core".  And we seem to need helpers like `appendSeeBlock()` for each property.

Perhaps a better design would be to deprecate `customBlocks` and replace it with a collection `blocks` that simply includes ALL the blocks.  Then helpers like `seeBlocks` would be a shorthand for something like this:
```ts
  public getBlocksOfType(tag: TSDocTagDefinition): ReadonlyArray<DocBlock> {
    return this._blocks.filter(x => x.blockTag.tagNameWithUpperCase === tag.tagNameWithUpperCase);
  }

  public get seeBlocks(): ReadonlyArray<DocBlock> {
    return this.getBlocksOfType(StandardTags.see);
  }
```

@hbergren Do you like this design better?  If so we can implement it in a separate PR.

Reply:
Looking more closely at the code, it seems that `customBlocks` is also used by [TSDocEmitter](https://github.com/microsoft/tsdoc/blob/dd262cab0e08cb5baff33d48d6021610a8e40fc0/tsdoc/src/emitters/TSDocEmitter.ts#L124) when rearranging the blocks into a normalized order.  The summary/remarks/params/returns are ordered first, then all the "custom blocks" can be emitted afterwards.

So maybe we don't need to deprecate `customBlocks`.  We can add `DocComment.blocks` alongside it.

Reply:
> Hmm... this API feels weird to me. Perhaps a better design would be to deprecate customBlocks and replace it with a collection blocks that simply includes ALL the blocks.

I think your proposed design is pretty solid. I'm happy to help implement that in a future PR.

How would that potentially work with `TSDocEmitter`? We ensure that the `this._blocks` property is ordered before it gets to `TSDocEmitter`?

Reply:
Let's address this issue in a separate PR. When I started to prototype it, it has some design questions of its own.  I am okay with temporarily introducing `appendSeeBlock()` to get this PR merged.  We can mark the API as `@internal` for now.